modulus/models/diffusion/layers.py

```diff
@@ -0,0 +1,541 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Model architecture layers used in the paper "Elucidating the Design Space of 
+Diffusion-Based Generative Models".
+"""
+
+from typing import Any, Dict, List
+
+import numpy as np
+import torch
+from torch.nn.functional import silu
+
+from modulus.models.diffusion import weight_init
+
+
+class Linear(torch.nn.Module):
```

Comment:
Is this function necessary as a standalone layer under `models/diffusion`. It seems like the only purpose for its existance is to support layer initialization. If so, then maybe we should include it under `models/layers`.

Reply:
Added to this epic: https://github.com/NVIDIA/modulus/issues/196